CanDIG/CanDIGv2
# Update candig-server deployment to use Opa

Set up the stack as described in docs/install-docker.md.

To test:

* Get a user token, where the values for the data parameters are found in the files in `tmp/secrets`:
```
curl -X "POST" "http://auth.docker.localhost:8080/auth/realms/candig/protocol/openid-connect/token" \
     -H 'Content-Type: application/x-www-form-urlencoded; charset=utf-8' \
     --data-urlencode "client_id=local_candig" \
     --data-urlencode "client_secret=<value in keycloak-client-local_candig-secret>" \
     --data-urlencode "grant_type=password" \
     --data-urlencode "username=<value in keycloak-test-user>" \
     --data-urlencode "password=<keycloak-test-user-password>" \
     --data-urlencode "scope=openid"
```
* Using the access_token from that, get the results from:
```
curl "http://docker.localhost:3001/datasets/search" \
     -H 'Authorization: Bearer <access_token>'
```

You should get a result that looks something like 
```
{"status": {"Known peers": 1, "Queried peers": 1, "Successful communications": 1, "Valid response": true}, "results": {"datasets": [{"id": "WyJtb2NrMSJd", "name": "mock1"}], "total": 1}}
```
(it contains the dataset "mock1")
